web/base_pdf_page_view.js

```diff
@@ -16,6 +16,12 @@
 import { RenderingCancelledException } from "pdfjs-lib";
 import { RenderingStates } from "./ui_utils.js";
 
+// We draw on the canvas at 60fps (in using `requestAnimationFrame`), so if the
+// canvas is large, updating it at 60fps can be a way too much and can cause
+// some serious performance issues.
+// To avoid that we only update the canvas every 500ms.
+const MIN_DURATION_TO_UPDATE_CANVAS = 500; // ms
```

Comment:
We could even increase this (e.g. to 1 sec), so users see that something is being drawn, but we don't spend too much resources to draw too often.